src/protocol/vm/state.rs

```
@@ -522,6 +536,24 @@ impl VMPoolState<PreCachedDB> {
 
         merged
     }
+
+    fn init_token_storage_slots(&mut self) {
+        for t in self.tokens.iter() {
+            if self.involved_contracts.contains(t) && !self.token_storage_slots.contains_key(t) {
+                self.token_storage_slots.insert(
+                    *t,
+                    token::brute_force_slots(
+                        t,
+                        &self.block,
+                        self.engine
+                            .as_ref()
+                            .expect("engine should be set"),
+                    )
+                    .unwrap_or_else(|_| panic!("Couldn't brute force slots for {}", t)),
```

I think we should return the error here like we do in the rest of the file.